src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -107,24 +109,147 @@ const columns = [
+  // This can be a candidate for its own component once backend feature implementation is complete.
+  const pageSizeOptions = [
+    { value: 1, label: '1' },
+    { value: 5, label: '5' },
+    { value: 10, label: '10' },
+    { value: 20, label: '20' },
+    { value: 50, label: '50' }
+  ];
+
+  // modifies state setters to reset page as well.
+  function setStateAndReset<T>(stateChanger: React.Dispatch<T>): React.Dispatch<T> {
+    return (value: T) => {
+      stateChanger(value);
+      resetPage();
+    };
+  }
+
+  // Converts the columnFilters array into backend query parameters.
+  // Memoized as derived data to prevent infinite re-rendering.
+  // TEMP IMPLEMENTATION. Values currently hardcoded with knowledge of what a ColumnFilter is.
+  // TODO: remove hardcoding conversion of all submissions to column filter. it is a hacky workaround.
+  // TODO: implement reversible backend-frontend name conversion for use in RequestsSaga and here, remove hardcode.
+  // TODO: make a controller component, like in the achievements page, to handle conversion of page state into JSON.
+  const backendFilterParams = useMemo(() => {
+    return columnFilters
+      .concat(
+        showAllSubmissions
+          ? []
+          : [
+              {
+                id: 'gradingStatus',
+                value: GradingStatuses.none
+              }
+            ]
+      )
+      .map((column: ColumnFilter) => {
+        // TODO: change all references to column properties in backend saga to backend name to reduce
+        // un-needed hardcode conversion, ensuring that places that reference it are updated.
+        switch (column.id) {
+          case 'assessmentName':
+            return { title: column.value };
+          case 'assessmentType':
+            return { type: column.value };
+          case 'studentName':
+            return { name: column.value };
+          case 'studentUsername':
+            return { username: column.value };
+          case 'submissionStatus':
+            return { status: column.value };
+          case 'groupName':
+            return { groupName: column.value };
+          case 'gradingStatus':
+            if (column.value === GradingStatuses.none) {
+              return {
+                isManuallyGraded: true,
+                status: 'submitted',
+                numGraded: 0
+              };
+            } else if (column.value === GradingStatuses.graded) {
+              // TODO: coordinate with backend on subquerying to implement the third query
+              // currently ignored by backend as of 16 Feb 24 commit
+              return {
+                isManuallyGraded: true,
+                status: 'submitted',
+                numGradedEqualToTotal: true
+              };
+            } else {
+              // case: excluded or grading. Not implemented yet.
+              return {};
+            }
+          default:
+            return column;
+        }
+      })
```

Comment:
This callback function seems like it can be moved outside the component, or perhaps even into `GradingUtils` to keep this table component small.

Reply:
Agreed, you could perhaps convert it into a helper function which takes in columnFilters and showAllSubmissions and remove the function from the component.